libs/blocks/global-navigation/blocks/profile/profile.js

```diff
@@ -9,87 +10,89 @@ function decorateEmail(email) {
+    this.profileButtonEl = profileButtonEl;
+    this.decoratedEl = decoratedEl;
+    this.init();
   }
 
-  if (sections.manage.items.enterprise?.id) {
-    const manageLink = blockEl.querySelector('div > div > p:nth-child(4) a');
-    manageLink.href = decorateProfileLink(manageLink.href, 'adminconsole');
-    const manageEnt = createTag('li', { class: 'gnav-profile-action' }, manageLink);
-    profileActions.append(manageEnt);
+  async init() {
+    const { displayName, email } = await window.adobeIMS.getProfile();
+    if (this.profileButtonEl) this.profileButtonEl.setAttribute('aria-label', displayName);
+    this.displayName = displayName;
+    this.email = email;
+    this.decoratedEl.append(this.menu());
+    this.decoratedEl.addEventListener('click', () => this.toggleMenu(this.decoratedEl));
+    this.decoratedEl.dispatchEvent(new Event('profile_ready'));
   }
 
-  const signOutLink = blockEl.querySelector('div > div > p:nth-child(5) a');
-  signOutLink.addEventListener('click', (e) => {
-    e.preventDefault();
-    window.adobeIMS.signOut();
-  });
-  const signOut = createTag('li', { class: 'gnav-profile-action' }, signOutLink);
-  profileActions.append(signOut);
-
-  profileHeader.append(profileImg, profileDetails);
-  profileMenu.append(profileHeader, profileActions);
-  profileEl.append(profileButton, profileMenu);
-}
-
-async function profile(blockEl, profileEl, toggle, ioResp) {
-  const gnav = profileEl.closest('nav.gnav');
-  gnav.classList.add('signed-in');
+  decorateSignOut() {
+    const signOutLink = toFragment`<li class="gnav-profile-action">${this.signOutEl}</li>`;
+    signOutLink.addEventListener('click', (e) => {
+      e.preventDefault();
+      window.adobeIMS.signOut();
+    });
+    return signOutLink;
+  }
 
-  const profiles = {};
-  profiles.ims = await window.adobeIMS.getProfile();
-  profiles.io = await ioResp.json();
-  decorateProfileMenu(blockEl, profileEl, profiles, toggle);
+  menu() {
+    return toFragment`
+      <div id="gnav-profile-menu" class="gnav-profile-menu">
+        <a 
+          href="${decorateProfileLink(this.accountLinkEl.href, 'account')}" 
+          class="gnav-profile-header"
+          aria-label="${this.accountLinkEl.textContent}"
+        >
+          ${this.avatarImgEl.cloneNode(true)}
+          <div class="gnav-profile-details">
+            <p class="gnav-profile-name">${this.displayName}</p>
+            <p class="gnav-profile-email">${decorateEmail(this.email)}</p>
+            <p class="gnav-profile-account">${this.accountLinkEl.innerHTML}</p>
+          </div>
+        </a>
+        <ul class="gnav-profile-actions">
+          ${this.sections.manage.items.team?.id ? decorateAction(this.manageTeamsEl) : ''}
+          ${this.sections.manage.items.enterprise?.id ? decorateAction(this.manageEnterpriseEl) : ''}
+          ${this.decorateSignOut()}
+        </ul>
+      </div>
+    `;
+  }
 }
-
-export default { profile };
+export default { Profile };
+export { Profile };
```

Comment:
Are both exports needed?

Reply:
Nope, not if my logic gets a little smarter 😁 changed!